### ISSUE TITLE
Different loss than the original paper

### ISSUE THREAD

**Issue by gaohuazuo** [carpedm20/DCGAN-tensorflow#33]
The generator loss in the original GAN paper is

<img src="https://latex.codecogs.com/gif.latex?\log\left(1&space;-&space;{1&space;\over&space;{1&space;&plus;&space;e^{-x}}}\right)&space;=&space;-&space;\log(1&space;&plus;&space;e^x)" title="\log\left(1 - {1 \over {1 + e^{-x}}}\right) = - \log(1 + e^x)" />

The generator loss implemented in this repo is

<img src="https://latex.codecogs.com/gif.latex?-\log\left(1&space;\over&space;{1&space;&plus;&space;e^{-x}}\right)&space;=&space;\log(1&space;&plus;&space;e^{-x})" title="-\log\left(1 \over {1 + e^{-x}}\right) = \log(1 + e^{-x})" />

**Comment by dtward** [comment]
Hi gaohuazuo, from the paper here: https://arxiv.org/pdf/1406.2661v1.pdf , right below equation (1) which defines the loss function, they state

"Rather than training G to minimize log(1-D(G(z))) we can train G to maximize log(D(G(z))).  This objective function results in the same fixed point of the dynamics of G and D but provides much stronger gradients early in learning"

If you want to minimize equation (1) directly, you can just write

    self.g_loss = self.d_loss_fake*(-1.0)

instead of

    self.g_loss = tf.reduce_mean(tf.nn.sigmoid_cross_entropy_with_logits(self.D_logits_, tf.ones_like(self.D_)))

In my experience there's not a big difference.

**Comment by carpedm20** [comment]
Thanks for the discussion and yes I think the equation is same.
